lib/tasks/school_urn_generator.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+ALL_URNS = (1..999_999).freeze unless defined?(ALL_URNS)
+
+class SchoolURNGenerator
+  class << self
+    def next
+      sprintf("%06d", taken.push(available.pop || (raise "URN available list exhausted"))[-1])
```

Comment:
There is like 4 or 5 things in that line. Please do it across multiple lines. In particular I don't know what `[-1]` is doing here. I think it is only facilitating sticking all of it in one line and avoiding a local variable, which I reckon is not a good enough reason to justify itself.

Reply:
Also, all comments in this file relate to TRN validation too. I think I approved an old, and pretty different version of it, sorry I didn't take another look.